examples/custom_rendering_wgpu/src/controls.rs

```diff
@@ -0,0 +1,368 @@
+//! This example showcases a widget that uses a custom shader to be rendered.
+use std::cell::Cell;
+
+use iced_graphics::{Backend, Defaults, Primitive, Renderer};
+use iced_native::{
+    event, layout, mouse, Background, Clipboard, Color, Element, Event, Hasher,
+    Layout, Length, Point, Rectangle, Size, Widget,
+};
+use iced_wgpu::wgpu;
+
+use bytemuck::{Pod, Zeroable};
+
+use iced_graphics::Container;
+use iced_graphics::Space;
+use iced_winit::{slider, Align, Column, Command, Program, Row, Slider, Text};
+
+pub struct CustomRenderedWidget<'a> {
+    size: f32,
+    bounds: &'a Cell<Rectangle>,
+}
+
+impl<'a> CustomRenderedWidget<'a> {
+    pub fn new(size: f32, bounds: &'a Cell<Rectangle>) -> Self {
+        Self { size, bounds }
+    }
+}
+
+impl<'a, Message, B> Widget<Message, Renderer<B>> for CustomRenderedWidget<'a>
+where
+    B: Backend,
+{
+    fn width(&self) -> Length {
+        Length::Shrink
+    }
+
+    fn height(&self) -> Length {
+        Length::Shrink
+    }
+
+    fn layout(
+        &self,
+        _renderer: &Renderer<B>,
+        _limits: &layout::Limits,
+    ) -> layout::Node {
+        layout::Node::new(Size::new(self.size, self.size))
+    }
+
+    fn on_event(
+        &mut self,
+        _event: Event,
+        _layout: Layout<'_>,
+        _cursor_position: Point,
+        _renderer: &Renderer<B>,
+        _clipboard: &mut dyn Clipboard,
+        _messages: &mut Vec<Message>,
+    ) -> event::Status {
+        event::Status::Ignored
+    }
+
+    fn hash_layout(&self, state: &mut Hasher) {
+        use std::hash::Hash;
+
+        self.size.to_bits().hash(state);
+    }
+
+    fn draw(
+        &self,
+        _renderer: &mut Renderer<B>,
+        _defaults: &Defaults,
+        layout: Layout<'_>,
+        _cursor_position: Point,
+        _viewport: &Rectangle,
+    ) -> (Primitive, mouse::Interaction) {
+        // We update the current bounds of this custom widget, so that the custom rendering
+        // step will render it in the correct spot
+        self.bounds.replace(layout.bounds());
```

Comment:
Although the full example is pretty long, this is the key point, where the `layout` is extracted to locate the custom render pass.